src/full/Agda/Interaction/Options/Base.hs

```diff
@@ -1656,6 +1685,15 @@ pragmaOptions = concat
   , pragmaFlag      "sized-types" lensOptSizedTypes
                     "enable sized types" "(inconsistent with --guardedness)"
                     $ Just "disable sized types"
+  , pragmaFlag      "type-based-termination" lensOptTypeBasedTermination
+                    "enable type-based termination checker" "(ignored if used together with --guardedness, --sized-types)"
```

Comment:
Ignoring the flag seems like poor practice, we should rather have an error if there is an incompatibility between flags.